k8s/helm-chart/dynatrace-gcp-function/templates/active-gate-pod.yaml

```diff
@@ -0,0 +1,108 @@
+# Copyright 2021 Dynatrace LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+{{- if or (eq .Values.deploymentType "logs") (eq .Values.deploymentType "all") }}
+  {{- if (.Values.activeGate.autoDeploy) }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: dynatrace-activegate-gcpmon-router
+  namespace: dynatrace
+spec:
+  type: ClusterIP
+  selector:
+    app: dynatrace-ag-gcpmon
+  ports:
+    - protocol: TCP
+      port: {{ .Values.activeGate.autoDeployPort }}
+      targetPort: ag-https
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: dynatrace-gcpmon-pullsecret
+  namespace: dynatrace
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: {{ template "imagePullSecret" . }}
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: dynatrace-ag-gcpmon
+  labels:
+    app: dynatrace-ag-gcpmon
+  namespace: dynatrace
+spec:
+  selector:
+    matchLabels:
+      app: dynatrace-ag-gcpmon
+  replicas: 1
+  serviceName: dynatrace-ag-gcpmon
+  template:
+    metadata:
+      labels:
+        app: dynatrace-ag-gcpmon
+        envid: {{ template "environmentID" . }}
+      annotations:
+        "version": "1"
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+        checksum/secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+    spec:
+      containers:
+        - name: dynatrace-ag-gcpmon
+          image: {{ template "activeGateImage" . }}
+          imagePullPolicy: Always
+          ports:
+            - name: ag-https
+              containerPort: 9999
+          env:
```

Comment:
I'm wondering... if we had two AGs installed in the same gcp project (image downloaded from the same tenant), they would have the same id?

Reply:
that is corner case but still possible, @mswiatkowska do you remember how it is handled for Azure logs forwared? if I remember correctly Azure Resource Group name is used as one of SEED components?

Reply:
I've added a .Values.clusterName to generate a new ID